### ISSUE TITLE
Stratum spec violation

### ISSUE THREAD

**Issue by sammy007** [nicehash/nheqminer#3]
What is an idea behind adding `mining.extranonce.subscribe`?

https://github.com/str4d/zips/blob/77-zip-stratum/drafts/str4d-stratum/draft1.rst

There is big work done on specification.

Protocol Flow

Client sends mining.subscribe to set up the session.
Server replies with the session information.
Client sends mining.authorize for their worker(s).
Server replies with the result of authorization.
Server sends mining.set_target.
Server sends mining.notify with a new job.
Client mines on that job.
Client sends mining.submit for each solution found.
Server replies with whether the solution was accepted.
Server sends mining.notify again when there is a new job.

**Comment by nicehashdev** [comment]
We added mining.extranonce.subscribe as an extension for compatibility with our service. Without this extension miner would still work but would get disconnected every time it is swapped to another order. We created this extension for original bitcoin stratum protocol long time ago and have been using it for 2 years. Most of miner software including ASIC supports it. With extension miners do not get disconnect thus perform better.

Since this is a miner for NiceHash, I am not fully sure where exactly is the problem?
